core/src/parser/parser.rs

```diff
@@ -55,13 +55,52 @@ impl Parser {
     false
   }
 
+  pub fn is_tok_curr(&self, tok: Token, curr: Option<&Token>) -> bool {
```

Comment:
rename to is_tok_eq if what you want to do is compare tokens, otherwise remove curr and get it from self instead